status-reporter/status_reporter/status_reporter.py

```diff
@@ -0,0 +1,139 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import os
+import time
+import json
+import base64
+import random
+import logging
+from logging.handlers import WatchedFileHandler
+
+import yaml
+import requests
+
+from .constants import CONFIGURATION_PATH, STATUS_REPORTER
+
+
+LOGFILE = '/var/log/cloudify/status-reporter/reporter.log'
+
+logger = logging.getLogger(STATUS_REPORTER)
+logger.setLevel(logging.INFO)
+formatter = logging.Formatter(fmt='%(asctime)s [%(levelname)s] '
+                                  '[%(name)s] %(message)s',
+                              datefmt='%d/%m/%Y %H:%M:%S')
+file_handler = WatchedFileHandler(filename=LOGFILE)
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+
+class InitializationError(Exception):
+    pass
+
+
+class Reporter(object):
+    def __init__(self, sampler, component_type):
+        if not callable(sampler):
+            raise InitializationError('Reporter expected a callable sampler,'
+                                      ' got {0!r}..')
+        self.status_sampler = sampler
+
+        try:
+            with open(CONFIGURATION_PATH) as f:
+                self._config = yaml.safe_load(f)
+        except yaml.YAMLError as e:
+            raise InitializationError('Failed loading status reporter\'s '
+                                      'configuration with the following: {0}'.format(e))
+
+        self._reporter_user_name = self._config.get('user_name', None)
+        self._reporter_token = self._config.get('token', None)
+        self._ca_path = self._config.get('ca_path', None)
+        self._managers_ips = self._config.get('managers_ips', [])
+
+        if not all([self._managers_ips,
+                    self._reporter_token,
+                    self._reporter_user_name
+                    ]):
+            invalid_conf_settings = {
+                'Managers Ips': self._managers_ips,
+                'Username': self._reporter_user_name,
+                'Password token': self._reporter_token
+            }
+            raise InitializationError('Please verify the reporter\'s config related to '
+                                      '{0} ..'.format(json.dumps(invalid_conf_settings,
+                                                                 indent=1)))
+
+        if self._ca_path and os.path.exists(self._ca_path):
+            raise InitializationError('CA certificate was not found, '
```

Comment:
If we have a ca certificate and the path exists then we complain that we can't find it? I think this might want to be the other way around?

Reply:
forgot the not ;)